vignettes/join-keys.Rmd

```diff
@@ -10,37 +12,95 @@ vignette: >
+)
+
+# The parent of ds2 and ds3 is ds1
+# converts relationship to child-parent
+#  ds1 <--> ds2 becomes ds1 <-- ds2
+#  ds1 <--> ds3 becomes ds1 <-- ds3
+parents(jk) <- list(ds2 = "ds1", ds3 = "ds1")
+
+jk
+```
+
+| Output of `print(jk)`              | Output annotation                    |
+| ---------------------------------- |:----------------------------------------:|
+| `## A join_keys object containing foreign keys between 3 datasets:` |  **Title** |
+| `## ds1: [id1]`                |  **Primary keys**<br />_\<dataset name\>_: [_\<primary_keys\>_]                                          | 
+| `##   <-- ds2: [id1]`          | **Foreign keys**<br /> _(arrow `<--` denotes `ds1` is the parent of `ds2`)_                    | 
+| `##   <-- ds3: [id1]`          |                                                                               | 
+| `## ds2: [id1, id2]`           |                                                                               | 
+| `##   --> ds1: [id1]`          | arrow `-->` denotes `ds2` is a child of `ds1`                                                                              | 
+| `##   --* (implicit via parent with): ds3` |  **Implicit relationship between `ds2` & `ds3`**<br />_(given that they share common keys with same parent)_    |
+| `## ds3: [id1, id3]`           |                                                                               | 
+| `##   --> ds1: [id1]`          |                                                                               | 
+| `##   --* (implicit via parent with): ds2` |                                                                               | 
+| `## ds4: [no primary keys]`    |                                                                               | 
+| `##   <-> ds5: [id5]`          |  **Foreign keys** <br />_(arrow `<->` denotes no parent definition between datasets)_      | 
+| `## ds5: [no primary keys]`    |                                                                               | 
+| `##   <-> ds4: [id4] `         |                                                                               | 
+
+### Primary key with `teal_data`
+
+The simplest method to define the join keys when calling the `teal_data` function is to use the `join_keys` argument.
+We can specify the column(s) of the dataset that _(together)_ uniquely identify rows in the dataset.
+
+```{r, include=FALSE}
+# nolint start: commented_code_linter.
+```
 
 ```{r, message=FALSE}
 library(teal.data)
 library(dplyr)
 
 iris_with_keys <- mutate(iris, id = factor(row_number()))
 
-ds <- dataset(
-  "IRIS",
-  iris_with_keys,
+td <- teal_data(
+  IRIS = iris_with_keys,
   code = "IRIS <- iris %>% mutate(id = factor(row_number()))",
-  keys = c("id")
+  join_keys = join_keys(join_key("IRIS", keys = "id"))
+
+  ## Alternatively, when only a single primary key is defined the `join_keys()`
+  ##  wrapper can be skipped
```

Comment:
```suggestion
  ## wrapper can be skipped
```

Reply:
I kinda like the second lines of comments to have the extra space, but that is arbitrary and personal.

Implemented it in 07982451. Thanks!